### PR TITLE
Upstream master kflang

### LINKED ISSUE #55: Implement the Language Env Variable
This issue use to be part of https://github.com/StatCan/kubeflow/issues/49 but was separated for easier tracking. 

The commits related to this feature that need to be integrated into upstream code are:
- Jan 21: feat: Add Env variable to set language of the notebook server (StatCan/jupyter-apis@83af673)

This issue is for the process of creating the PR upstream. 
It will be needed to create the python equivalent of our go code modification. 
Another issue should then be created to keep track of that PR until it is merged upstream.

### PR DESCRIPTION
**Pull request by rohank07**

### CODE REVIEW

File: components/crud-web-apps/jupyter/backend/apps/common/form.py [hunk 1]
```diff
@@ -333,3 +336,9 @@ def add_notebook_volume(notebook, vol_name, claim, mnt_path):
     # Container Mounts
     mnt = {"mountPath": mnt_path, "name": vol_name}
     container["volumeMounts"].append(mnt)
+
+def set_notebook_language(notebook, body, defaults):
+    lang = get_form_value(body, defaults, "language")
+    if not re.match('^[a-zA-Z]{2}(_[a-zA-Z]{2})?$', lang):
+        raise BadRequest("Error: the value of KF_LANG environment variable ('" + lang + "') is not a valid format (e.g 'en', 'en_US', ...)")
```
Comment (wg102) [inline components/crud-web-apps/jupyter/backend/apps/common/form.py:343]: This message needs to be i18n. Look at how other messages are i18n within the code for help.

File: components/crud-web-apps/jupyter/frontend/src/app/pages/form/form-default/form-advanced-options/form-advanced-options.component.html [hunk 2]
```diff
@@ -1,9 +1,25 @@
...
+    <div class="collapsible-content">
+      <div class="content-inner">
+        <mat-slide-toggle formControlName="shm" class="matSpacing">
+          {{ "jupyter.formAdvancedOptions.toggleSharedMemory" | translate }}
+        </mat-slide-toggle>
+        <mat-form-field class="wide" appearance="outline">
+        <mat-label>{{ "jupyter.formAdvancedOptions.lblSystemLanguage" | translate }}</mat-label>
+        <mat-select matNativeControl  formControlName="language">
+          <mat-option *ngFor="let v of languageList" [value]="v.id">
+            {{ v.label | translate }}
```
Comment (wg102) [inline components/crud-web-apps/jupyter/frontend/src/app/pages/form/form-default/form-advanced-options/form-advanced-options.component.html:18]: The use of the variable v, maybe a better name would be more appropriate

File: components/crud-web-apps/jupyter/frontend/src/app/pages/form/form-default/form-advanced-options/form-advanced-options.component.scss [hunk 3]
```diff
@@ -1,3 +1,59 @@
...
+.collapsible-content .content-inner {
+  border: 1px solid rgba(224, 224, 224, 0.45);
+  border-bottom-left-radius: 7px;
+  border-bottom-right-radius: 7px;
+  padding: .5rem 1rem;
+}
+
+.matSpacing {
+  padding: 0.6em 0em;
+}
```
Comment (wg102) [inline components/crud-web-apps/jupyter/frontend/src/app/pages/form/form-default/form-advanced-options/form-advanced-options.component.scss:59]: Add end of file newline

File: components/crud-web-apps/jupyter/frontend/src/app/pages/form/form-default/form-default.component.ts [hunk 4]
```diff
@@ -155,6 +155,10 @@ export class FormDefaultComponent implements OnInit, OnDestroy {
       notebook.workspace.size = notebook.workspace.size.toString() + 'Gi';
     }
 
+    if (typeof notebook.language === 'string') {
+      notebook.language = notebook.language.toString()
+    }
```
Comment (wg102) [inline components/crud-web-apps/jupyter/frontend/src/app/pages/form/form-default/form-default.component.ts:160]: Can it be anything else than a string? Why the verification?

**Review by wg102 — CHANGES_REQUESTED** [review]

**Comment by rohank07** [comment]
Closing PR. Upstream replaced ngx-translate with localize. Rather than rebasing this branch (conflicts), a new branch is created to rebase.

**Pull request closed without merging**